src/tools/dev_server/hot-reload-server.ts

```diff
@@ -48,13 +48,33 @@ export class HotReloadServer {
         const files = JSON.parse(msg.toString());
         for (const file of files) {
           this.filesToWatch.push(file);
-          const local = file.replace(/^https:\/\/\$particles\//, './particles/');
+          let local: string = file.replace(/^https:\/\/\$particles\//, './particles/');
+          local = local.replace(/^https:\/\/\$arcs\//, './');
 
           console.log(`Watching: ${local}`);
-          this.watchers.push(chokidar.watch(local).on('change', path => {
+          let watcher: chokidar.FSWatcher;
+          if (local.endsWith('.wasm')) {
+            watcher = chokidar.watch(local, {
+              awaitWriteFinish: {
```

Comment:
Ok, now I see you have built end-of-write detection into our watcher :)

Does this mean the delay loop in loading the wasm file is just a backup for this detection not working? If so, we would want to instigate that only when we know there's been a reload event (so not on initial load), and still put in a spin limit (the reloaded file could be still be malformed).